activesupport/lib/active_support/cache.rb

```diff
@@ -687,7 +681,7 @@ def expanded_key(key)
               expanded_key(key.first)
             end
           when Hash
-            key.collect { |k, v| "#{k}=#{v}" }.sort
+            key.collect { |k, v| "#{k}=#{v}" }.sort!
```

Comment:
Really wish Ruby's interpreter would understand direct chaining to automatically mutate immediately altered collections.

`arr.map { }.compact.select().uniq.sort` why do I have to change my code? 😄

Reply:
That would be amazing. I wonder how hard it'd be to implement that into Ruby and if there'd be any gotchas.